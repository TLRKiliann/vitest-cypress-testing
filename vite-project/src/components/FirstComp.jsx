import { useState, useRef, useEffect } from 'react'
import ThirdComp from './ThirdComp'

let countId = 0;

export default function FirstComp({showing, setShowing}) {
  
  const [inputValue, setInputValue] = useState("")
  const [display, setDisplay] = useState([])
  const date = new Date()

  const handleSubmit = (e) => {
    e.preventDefault()
    if (inputValue) {
      setDisplay([...display, {date, id: countId++, name: inputValue, editor: false}])
    }
    setInputValue("")
  }

  const handleEventChange = (e) => {
    setInputValue(e.target.value)
  }

  const handleReverse = () => {
    const reversed = display.reverse()
    setDisplay([...reversed])
  }

  return (
    <>
      <h1 
        data-testid="first-comp"
        style={showing ? {color: "cyan"} : {color: "hotpink"}}>
        React-Dev
      </h1>

      <form data-testid="testid-form" onSubmit={handleSubmit}>
        <input
          data-testid="testid-event"
          type="text"
          value={inputValue}
          onChange={(e) => handleEventChange(e)}
        />
        <button
          data-testid="testid"
          type="submit">
          Enter
        </button>
      </form>

      <button
        data-testid="testid-reverse"
        type="button"
        onClick={handleReverse}>
        Reverse
      </button>

      {display.map((dis) => (
        <ThirdComp
          key={dis.id}
          inputValue={inputValue}
          dis={dis}
          name={dis.name}
          editor={dis.editor}
          display={display}
          setDisplay={setDisplay}
        />
      ))}
    </>
  )
}