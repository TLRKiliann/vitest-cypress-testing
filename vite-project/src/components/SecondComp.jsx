import { useState, useEffect } from 'react'

const SecondComp = ({showing, setShowing}) => {

  const [person, setPerson] = useState({
    firstName: "Henry",
    password: "Yohout"
  })
  
  const [isLoading, setIsLoading] = useState(false)

  useEffect(() => {
    setIsLoading(true)
    console.log("this will finish in 3 sec")
    const timer = setTimeout(() => {
      console.log("run into setTimeout(fn())")
    }, 3000)
    setIsLoading(false)
    return () => clearTimeout(timer)
  }, [])

  const [count, setCount] = useState(0)

  const handleChangeFirstName = (e) => {
    setPerson({...person, firstName: e.target.value})
  }

  const handleChangePassword = (e) => {
    setPerson({...person, password: e.target.value})
  }

  const handleChange = () => {
    setShowing(!showing)
    setCount(count => count + 1)
  }

  return (
    <>
      <h1
        data-testid="second-comp" 
        style={showing ? {color: "cyan"} : {color: "hotpink"}}>
        SecondComp
      </h1>

      {isLoading ? (
        <h2 
          data-testid="testid-hello" 
          style={{fontSize: "4rem"}}>Hello {person.firstName} !!!
        </h2>
        ) : (
          <>
          </>
        )
      }
      <div
        style={{
          display: 'flex',
          flexDirection: 'column',
          alignItems: "left",
          justifyContent: 'left',
          marginBottom: "20px"
        }}
      >
        <label style={{textAlign: "left"}}>username</label>
        <input
          data-testid="testid-inputone"
          name="username"
          type="text"
          value={person.firstName}
          onChange={(e) => handleChangeFirstName(e)}
          placeholder={person.firstName}
        />

        <label style={{textAlign: "left"}}>password</label>
        <input
          data-testid="testid-inputtwo"
          type="password"
          value={person.password}
          onChange={(e) => handleChangePassword(e)}
          required
        />
        
        <button
          data-testid="testid-two"
          type="button"
          onClick={handleChange}
        >
          {showing ? 'Show password' : 'Show firstname'} - {count}
        </button>
        {showing ? (
          <p>firstname: {person.firstName}</p>
          ) : (
          <p>password: {person.password}</p>
          )
        }
      </div>
    </>
  )
}
export default SecondComp;