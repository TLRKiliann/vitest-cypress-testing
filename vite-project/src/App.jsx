import { useState } from 'react'
import FirstComp from './components/FirstComp'
import SecondComp from './components/SecondComp'
import './App.css'

function App() {
  
  const [showing, setShowing] = useState(false)
  
  return (
    <>
      <FirstComp
        showing={showing}
        setShowing={setShowing}
      />
      <SecondComp
        showing={showing}
        setShowing={setShowing}
      />
    </>
  )
}

export default App