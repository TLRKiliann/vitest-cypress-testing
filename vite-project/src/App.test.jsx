import React from 'react'
import { test, expect, vi } from 'vitest'
import { screen, render } from '@testing-library/react'
import { create } from "react-test-renderer"
import "@testing-library/jest-dom"
import App from './App.jsx'
import FirstComp from './App.jsx'
import SecondComp from './App.jsx'

/*
test("mocking useState", () => {
  vi.mock("./App.jsx", () => {
    const actual = vi.importActual("./App.jsx")
    return actual;
  })
});
*/

test('MatchSnapShot test Subscribe', () => {
  const tree = create(
  	<App />
  )
  expect(tree.toJSON()).toMatchSnapshot()
});

test('FirstComp test', () => {
  const { getByTestId } = render(
    <FirstComp data-testid="testid-form" />
  );
  const testFirstId = screen.getByTestId("testid-form");
  expect(testFirstId).toBeInTheDocument();
});

test('SecondComp test', () => {
  const { getByTestId } = render(
    <SecondComp data-testid="testid-inputone" />
  );
  const testFirstId = screen.getByTestId("testid-inputone");
  expect(testFirstId).toBeInTheDocument();
});

test('passes setShowing prop to FirstComp and SecondComp', () => {
  render(
    <App />
  );
  const firstComp = screen.getByTestId('first-comp');
  const secondComp = screen.getByTestId('second-comp');
  expect(firstComp.getAttribute('setShowing')).toBeNull();
  expect(secondComp.getAttribute('setShowing')).toBeNull();
});