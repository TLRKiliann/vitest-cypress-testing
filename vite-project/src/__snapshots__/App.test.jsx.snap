// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MatchSnapShot test Subscribe 1`] = `
[
  <h1
    data-testid="first-comp"
    style={
      {
        "color": "hotpink",
      }
    }
  >
    React-Dev
  </h1>,
  <form
    data-testid="testid-form"
    onSubmit={[Function]}
  >
    <input
      data-testid="testid-event"
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      data-testid="testid"
      type="submit"
    >
      Enter
    </button>
  </form>,
  <button
    data-testid="testid-reverse"
    onClick={[Function]}
    type="button"
  >
    Reverse
  </button>,
  <h1
    data-testid="second-comp"
    style={
      {
        "color": "hotpink",
      }
    }
  >
    SecondComp
  </h1>,
  <div
    style={
      {
        "alignItems": "left",
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "left",
        "marginBottom": "20px",
      }
    }
  >
    <label
      style={
        {
          "textAlign": "left",
        }
      }
    >
      username
    </label>
    <input
      data-testid="testid-inputone"
      name="username"
      onChange={[Function]}
      placeholder="Henry"
      type="text"
      value="Henry"
    />
    <label
      style={
        {
          "textAlign": "left",
        }
      }
    >
      password
    </label>
    <input
      data-testid="testid-inputtwo"
      onChange={[Function]}
      required={true}
      type="password"
      value="Yohout"
    />
    <button
      data-testid="testid-two"
      onClick={[Function]}
      type="button"
    >
      Show firstname
       - 
      0
    </button>
    <p>
      password: 
      Yohout
    </p>
  </div>,
]
`;
